nixos/modules/services/matrix/mautrix-whatsapp.nix

```
@@ -0,0 +1,137 @@
+{ lib, config, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.mautrix-whatsapp;
+  dataDir = "/var/lib/mautrix-whatsapp";
+  format = pkgs.formats.json {};
+
+  registrationFile = "${dataDir}/whatsapp-registration.yaml";
+  settingsFile = format.generate "config.json" cfg.settings;
```

```suggestion
  settingsFormat = pkgs.formats.json {};

  registrationFile = "${dataDir}/whatsapp-registration.yaml";
  settingsFile = settingsFormat.generate "config.json" cfg.settings;
```
Let's call this `settingsFormat`. It's more descriptive and this is also the name of this variable in `mautrix-telegram.nix`.